app/services/host_inventory_api.rb

```diff
@@ -24,6 +24,14 @@ def host_already_in_inventory(hostname_or_id)
     find_results(JSON.parse(response.body))
   end
 
+  def inventory_host
+    @inventory_host ||= host_already_in_inventory(@id) ||
+                        (@hostname && host_already_in_inventory(@hostname))
```

Comment:
We can/should remove this find by hostname. If we don't have ID, we fail to parse.